compiler/rustc_session/src/filesearch.rs

```diff
@@ -62,9 +62,68 @@ pub fn make_target_lib_path(sysroot: &Path, target_triple: &str) -> PathBuf {
     PathBuf::from_iter([sysroot, Path::new(&rustlib_path), Path::new("lib")])
 }
 
-/// This function checks if sysroot is found using env::args().next(), and if it
-/// is not found, uses env::current_exe() to imply sysroot.
-pub fn get_or_default_sysroot() -> PathBuf {
+pub struct SysrootCompileTimeArgsForHost {
+    pub sysroot: Option<String>,
+    pub rustup_home: Option<String>,
+    pub multirust_home: Option<String>,
+    pub rustup_toolchain: Option<String>,
+    pub multirust_toolchain: Option<String>,
```

Comment:
FWIW given how long it has been since multirust stopped being developed, I don't think that is still worth supporting.